records/labels/spec.go

```diff
@@ -0,0 +1,58 @@
+package labels
+
+import (
+	"fmt"
+)
+
+const (
+	// SeparatorChar delimits concatenated labels.
+	SeparatorChar = '.'
+
+	DNS952MaxLength  int = 24
+	DNS1123MaxLength int = 63
+
+	// see http://www.rfc-base.org/txt/rfc-952.txt
+	HostNameSpecType952 HostNameSpecType = iota
+
+	// see http://www.rfc-base.org/txt/rfc-1123.txt
+	HostNameSpecType1123
```

Comment:
I believe this can be further simplified. No need for `HostNameSpecType`.
```go
var (
    RFC952   = HostNameSpec{table:  makeDNS952Table(), maxlen: 24}
    RFC1123  = HostNameSpec{table:  makeDNS1123Table(), maxlen: 63}
)
```